lgn-worker/src/main.rs

```diff
@@ -108,13 +108,18 @@ fn main() -> anyhow::Result<()> {
 
     let config = Config::load(cli.config);
     config.validate();
-    info!("Loaded configuration: {:?}", config);
+    info!("Loaded configuration. config: {:?}", config);
 
-    metrics_exporter_prometheus::PrometheusBuilder::new()
+    if let Err(err) = metrics_exporter_prometheus::PrometheusBuilder::new()
         .with_http_listener(([0, 0, 0, 0], config.prometheus.port))
-        .install()?;
+        .install()
+    {
+        error!("Creating prometheus metrics failed. err: {:?}", err);
+    }
 
-    run(&config)
+    if let Err(err) = run(&config) {
+        error!("Service exiting with an error. err: {:?}", err);
+    }
```

Comment:
`run` returns a `Result`. When it was `Err` the error was printed using Rust's default error handler, which print the error to stdout, because we have `RUST_BACKTRACE=full` set, we would have multiple lines printed to the stdout, which would not be collected together.

Explicitly handling the error allows to use `tracing` logging, and groups the log lines together.